src/main/java/org/jabref/gui/exporter/ExportCustomizationDialogViewModel.java

```diff
@@ -0,0 +1,91 @@
+package org.jabref.gui.exporter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.SimpleListProperty;
+import javafx.collections.FXCollections;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.DialogService;
+import org.jabref.logic.exporter.TemplateExporter;
+import org.jabref.logic.journals.JournalAbbreviationLoader;
+import org.jabref.preferences.PreferencesService;
+
+public class ExportCustomizationDialogViewModel extends AbstractViewModel {
+
+    private final ListProperty<ExporterViewModel> exporters = new SimpleListProperty<>(FXCollections.observableArrayList());
+    private final ListProperty<ExporterViewModel> selectedExporters = new SimpleListProperty<>(FXCollections.observableArrayList());
+
+
+    private final PreferencesService preferences;
+    private final DialogService dialogService;
+    private final JournalAbbreviationLoader loader;
+
+    public ExportCustomizationDialogViewModel(PreferencesService preferences, DialogService dialogService, JournalAbbreviationLoader loader) {
+        this.preferences = preferences;
+        this.dialogService = dialogService;
+        this.loader = loader;
+        loadExporters();
+    }
+
+    private void loadExporters() {
+        List<TemplateExporter> exportersLogic = preferences.getCustomExportFormats(loader); //Var exportersLogic may need more descriptive name
+        for (TemplateExporter exporter : exportersLogic) {
+            exporters.add(new ExporterViewModel(exporter));
+        }
+    }
+
+    public void addExporter() {
+        CreateModifyExporterDialogView dialog = new CreateModifyExporterDialogView(null, dialogService, preferences,
+                                                                                   loader);
+        Optional<ExporterViewModel> exporter = dialogService.showCustomDialogAndWait(dialog);
+        if ((exporter != null) && exporter.isPresent()) {
+            exporters.add(exporter.get());
+        }
+    }
+
+    /**
+     * Open dialog to modify exporter, which is the same as adding ane exporter but passes in
+     * a non-empty Optional of ExporterViewModel and sets the result into exporters.
+     */
+
+    public void modifyExporter() {
+        CreateModifyExporterDialogView dialog;
+        ExporterViewModel exporterToModify = null;
+        try {
+            exporterToModify = selectedExporters.get(0);
+            dialog = new CreateModifyExporterDialogView(exporterToModify, dialogService, preferences, loader);
+        } catch (IndexOutOfBoundsException ex) {
```

Comment:
It's not clear. I put debug breakpoints on lines 56, 57, 59, and 60. I also put one in the line `viewModel.modifyExporter()` in `ExporterCustomizationDialogView`.  That one is the only one it stops at before it throws the exception, but I don't understand how that is possible, since there should be no intervening lines. Also I find it odd that it should throw the very same exception that I was trying to catch.

Reply:
You can try to make `selectedExporters` a normal array list.

Reply:
I don't think it has to do with `selectedExporters` because it returns at line 59 when none are selected. Something else may be going on.